network/p2p/libP2PHostWrapper.go

```diff
@@ -0,0 +1,124 @@
+package p2p
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/libp2p/go-libp2p"
+	"github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/host"
+	libp2pnet "github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/protocol"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	tptu "github.com/libp2p/go-libp2p-transport-upgrader"
+	"github.com/libp2p/go-libp2p/config"
+	"github.com/libp2p/go-tcp-transport"
+	"github.com/multiformats/go-multiaddr"
+	"github.com/rs/zerolog"
+)
+
+// libP2PHostWrapper is an internal data structure to this package that encapsulates
+// all components that p2p package affiliates with a libp2p host.
+// Purpose of encapsulation is to ease passing the entire libp2p host affiliated components
+// around.
+type libP2PHostWrapper struct {
+	connGater *connGater     // used to provide white listing
+	host      host.Host      // reference to the libp2p host (https://godoc.org/github.com/libp2p/go-libp2p-core/host)
+	pubSub    *pubsub.PubSub // reference to the libp2p PubSub component
+}
+
+// ConnectionGater returns `connGater` component of the libP2PHostWrapper.
+func (h libP2PHostWrapper) ConnectionGater() *connGater {
+	return h.connGater
+}
+
+// Host returns `host` component of the libP2PHostWrapper.
+func (h libP2PHostWrapper) Host() host.Host {
+	return h.host
+}
+
+// PubSub returns `pubsub` component of the libP2PHostWrapper.
+func (h libP2PHostWrapper) PubSub() *pubsub.PubSub {
+	return h.pubSub
+}
+
+// SetStreamHandler sets the stream handler of the host component in the libP2PHostWrapper.
+func (h *libP2PHostWrapper) SetStreamHandler(flowLibP2PProtocolID protocol.ID, handler libp2pnet.StreamHandler) {
+	h.host.SetStreamHandler(flowLibP2PProtocolID, handler)
+}
+
+// bootstrapLibP2PHost creates and starts a libp2p host as well as a pubsub component for it, and returns all in a
+// libP2PHostWrapper.
+// In case `allowList` is true, it also creates and embeds a connection gater in the returned libP2PHostWrapper, which
+// whitelists the `allowListAddres` nodes.
+func bootstrapLibP2PHost(ctx context.Context,
+	logger zerolog.Logger,
+	nodeAddress NodeAddress,
+	conMgr ConnManager,
+	key crypto.PrivKey,
+	allowList bool,
+	allowListAddrs []NodeAddress,
+	psOption ...pubsub.Option) (*libP2PHostWrapper, error) {
+
+	var connGater *connGater
+
+	sourceMultiAddr, err := multiaddr.NewMultiaddr(MultiaddressStr(nodeAddress))
+	if err != nil {
+		return nil, fmt.Errorf("failed to translate Flow key to Libp2p key: %w", err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("failed to translate Flow address to Libp2p multiaddress: %w", err)
```
I don't think we're dealing with keys here